src/python/nimbusml/pipeline.py

```diff
@@ -1693,6 +1695,122 @@ def getn(n):
                     "only fit(X) is allowed or the training becomes "
                     "ambiguous.")
 
+    @trace
+    def calculate_feature_contributions(self, X, y=None,
+                 evaltype='auto', group_id=None,
+                 weight=None,
+                 verbose=0,
+                 top=10,
+                 bottom=10,
+                 normalize=True,
+                 as_binary_data_stream=False, **params):
+        """
+        Apply transforms and test with the final estimator, return metrics
+        """
+        # start the clock!
+        start_time = time.time()
+        self.verbose = verbose
+
+        if not self._is_fitted:
```

Comment:
>_is_fitted [](start = 20, length = 10)

is this set when you load model from file?

Reply:
This is set when I load model with pickle or from file with load_model(). In the case of pickle, feature contribution works fine. In the case of load_model(), even though self._is_fitted, there are no steps in the pipeline and it throws "TypeError: No steps given."

---
In reply to: [303650300](https://github.com/microsoft/NimbusML/pull/186#discussion_r303650300) [](ancestors = 303650300)

Reply:
Works with model loaded from zip file too.

---
In reply to: [303677532](https://github.com/microsoft/NimbusML/pull/186#discussion_r303677532) [](ancestors = 303677532,303650300)